packages/gestalt/src/Image.tsx

```diff
@@ -134,6 +141,30 @@ export default class Image extends PureComponent<Props> {
     }
   }
 
+  refCallback = (node: HTMLImageElement | null) => {
+    const { _fixCompletedOnLoad } = this.props;
+    // For certain scenarios, such as server-side rendering, the image may already be loaded by the time the component is rendered resulting in the onLoad event not being triggered.
+    // To address these, we can use a ref callback and check whether the image is already loaded - if it is, we trigger the onLoad event manually.
+    if (_fixCompletedOnLoad && node?.complete && !this.onLoadCalled) {
```

Comment:
`!this.onLoadCalled `

so `handleLoad` when image loads and we set `this.onLoadCalled = true;`. Then, when it renders we set the ref and we check `onLoadCalled`

`handleLoad` gets sets manually if `this.onLoadCalled = true;`. 

"if it is, we trigger the onLoad event manually."

should `&& !this.onLoadCalled ` be `&& this.onLoadCalled `

Reply:
`!this.onLoadCalled` is here to guard against a scenario where the native `img` load callback is fired (and `onLoad` called) before the ref callback

i think the only situation where this could happen is:
- image component renders
- image request completes very quickly
- ref callback is triggered

i think this would be an edge case but wanted to play it safe